OrbitCore/SymbolHelper.cpp

```diff
@@ -41,118 +44,117 @@ std::vector<std::string> ReadSymbolsFile() {
+
+  std::set<path> search_paths;
+  for (const auto& directory : search_directories) {
+    search_paths.insert(directory / filename_dot_debug);
+    search_paths.insert(directory / filename_plus_debug);
+    search_paths.insert(directory / filename);
+  }
 
-    ErrorMessageOr<std::unique_ptr<ElfFile>> symbols_file = ElfFile::Create(symbols_file_path);
-    if (!symbols_file) continue;
-    if (!symbols_file.value()->HasSymtab()) continue;
+  LOG("Trying to find symbols for module: \"%s\"", module_path.string());
+  for (const auto& symbols_path : search_paths) {
+    if (!std::filesystem::exists(symbols_path)) continue;
 
-    // don't check build_id of module_path itself has symbols
-    // and requested build_id is empty.
-    if (build_id.empty() && symbols_file_path == module_path) {
-      return symbols_file_path;
+    const auto verification_result = VerifySymbolsFile(symbols_path, build_id);
+    if (!verification_result) {
+      LOG("Existing file \"%s\" is not the symbols file for module \"%s\", error: %s", symbols_path.string(),
+          module_path.string(), verification_result.error().message());
+      continue;
     }
 
-    // Otherwise check that build_id is not empty and correct.
-    if (symbols_file.value()->GetBuildId().empty()) continue;
-    if (symbols_file.value()->GetBuildId() != build_id) continue;
-
-    LOG("Found debug info for module \"%s\" -> \"%s\"", module_path, symbols_file_path);
-    return symbols_file_path;
+    LOG("Found debug info for module \"%s\" -> \"%s\"", module_path.string(), symbols_path.string());
+    return symbols_path;
   }
 
   return ErrorMessage(
-      absl::StrFormat("Could not find a file with debug symbols for module \"%s\"", module_path));
+      absl::StrFormat("Could not find a file with debug symbols for module \"%s\"", module_path.string()));
 }
 
-ErrorMessageOr<ModuleSymbols> FindSymbols(const std::string& module_path,
-                                          const std::string& build_id,
-                                          const std::vector<std::string>& search_paths) {
-  ErrorMessageOr<std::string> debug_info_file_path =
-      FindSymbolsFile(module_path, search_paths, build_id);
-
-  if (!debug_info_file_path) {
-    return debug_info_file_path.error();
-  }
-
-  ErrorMessageOr<std::unique_ptr<ElfFile>> elf_file_result =
-      ElfFile::Create(debug_info_file_path.value());
+}  // namespace
 
-  if (!elf_file_result) {
-    return ErrorMessage(absl::StrFormat("Failed to load debug symbols for \"%s\" from \"%s\": %s",
-                                        module_path, debug_info_file_path.value(),
-                                        elf_file_result.error().message()));
-  }
+SymbolHelper::SymbolHelper()
+    : symbols_file_directories_(ReadSymbolsFile()), cache_directory_(Path::GetCachePath()) {}
 
-  return elf_file_result.value()->LoadSymbols();
+ErrorMessageOr<path> SymbolHelper::FindSymbolsWithSymbolsPathFile(
```

Comment:
Why does this method exists, if it just forwards to another method?

Reply:
True, I now did move things here.